src/primitives/star.js

```diff
@@ -0,0 +1,79 @@
+const {degToRad} = require('../math/utils')
+
+const vec2 = require('../math/vec2')
+
+const geom2 = require('../geometry/geom2')
+
+// @see http://www.jdawiseman.com/papers/easymath/surds_star_inner_radius.html
+const getRadiusRatio = (vertices, density) => {
+  if (vertices > 0 && density > 1 && density < vertices / 2) {
+    return Math.cos(Math.PI * density / vertices) / Math.cos(Math.PI * (density - 1) / vertices);
+  }
+  return 0
+}
+
+const getPoints = (vertices, radius, startAngle, center) => {
+  var a1 = degToRad(startAngle)
+  var a = 2 * Math.PI / vertices
+
+  var points = []
+  for (var i = 0; i < vertices; i++) {
+    let point = vec2.fromAngleRadians(a * i + a1)
+    vec2.scale(point, radius, point)
+    vec2.add(point, center, point)
+    points.push(point)
+  }
+  return points
+}
+
+/** Construct a star from the given options.
+ * @see https://en.wikipedia.org/wiki/Star_polygon
+ * @param {Object} [options] - options for construction
+ * @param {Array} [options.center=[0,0]] - center of star
+ * @param {Number} [options.vertices=5] - number of vertices (P) on the star
+ * @param {Number} [options.density=2] - density (Q) of star
+ * @param {Number} [options.outerRadius=1] - outer radius of vertices
+ * @param {Number} [options.innerRadius=0] - inner radius of vertices, or zero to calculate
+ * @param {Number} [options.startAngle=0] - starting angle for first vertice
+ *
+ * @example
+ * let star1 = star({vertices: 8, outerRadius: 10}) // star with 8/2 density
+ * let star2 = star({vertices: 12, outerRadius: 40, innerRadius: 20}) // star with given radius
+ */
+const star = (options) => {
```

Comment:
errr , why do we need stars in 'primitives' ? that seems like something you would build using a polygon ?

Reply:
These are difficult to create by hand, and this shape has been on my list of 'things' to add for a long long time.

FYI, I stole the code from maker.js, and converted to JSCAD.

Reply:
@danmarshall may be interested in this code as well, as there are a few improvements.

Reply:
hmm I understand that point, but that still contradicts the notion of specialized, 'what you need only' that we are shooting for for V2 right ?

Reply:
I'd still like to keep 'star' as this is a pretty fundamental shape. The code is concise, as well as there's a matematical basis for the code.

I know that you don't want feature creep, but I would add this anway either before or after V2 is released.